observation_portal/common/configdb.py

```diff
@@ -334,13 +334,15 @@ def get_optical_elements(self, instrument_type: str) -> dict:
         Returns:
              Available optical elements
         """
-        optical_elements = {}
+        optical_elements = defaultdict(list)
+        optical_elements_tracker = defaultdict(set)
         for instrument in self.get_instruments(exclude_states=['DISABLED', ]):
             if instrument_type.upper() == instrument['science_camera']['camera_type']['code'].upper():
                 for optical_element_group in instrument['science_camera']['optical_element_groups']:
-                    optical_elements[optical_element_group['type']] = []
                     for element in optical_element_group['optical_elements']:
-                        optical_elements[optical_element_group['type']].append(element)
+                        if element['code'] not in optical_elements_tracker[optical_element_group['type']]:
+                            optical_elements_tracker[optical_element_group['type']].add(element['code'])
```

Comment:
This might work by just returning the dict of sets rather than the lists. The serializer usage just looks like it is using `in` so it should be fine, so its just the stuff passed to the instrument info view, which might work out fine too with a set vs list.

Reply:
The thing this is compiling looks something like `{'filters': [{'code': 'V', 'name': 'Bessell-V'}, ... ]}`. Each container for the various optical element types contains a collection of dicts, and dicts are not hashable so they cannot go into a set. The instrument info view is used by the compose page, which needs more than just the code of each optical element.

Reply:
Oh yeah, I didn't realize that, this seems fine then